runtime/src/lib.rs

```diff
@@ -473,6 +480,29 @@ impl system::offchain::CreateTransaction<Runtime, UncheckedExtrinsic> for Runtim
 	}
 }
 
+/// Verify a Doughnut proof authorizes method dispatch given some input parameters
+impl additional_traits::DelegatedDispatchVerifier<CennznetDoughnut> for Runtime {
+	const DOMAIN: &'static str = "cennznet";
+
+	fn verify_dispatch(doughnut: &CennznetDoughnut, module: &str, method: &str) -> Result<(), &'static str> {
+		let mut domain = doughnut
+			.get_domain(Self::DOMAIN)
+			.ok_or("Doughnut does not grant permission for cennznet domain")?;
+		let cennznut: CENNZnutV0 = Decode::decode(&mut domain).map_err(|_| "Bad Doughnut encoding")?;
+
+		// Strips [c|p|s]rml- prefix
+		let module = cennznut
```

Comment:
Should use cennzut.validate() here (via trait Validate). Can just pass empty `args` for now `&[]` https://github.com/cennznet/cennznut-rs/blob/e09a993add8a831ecb574e068e730df76ac8c757/src/lib.rs#L29